arctic/store/_ndarray_store.py

```diff
@@ -510,13 +572,21 @@ def _do_write(self, collection, version, symbol, item, previous_version, segment
         # chunk and store the data by (uncompressed) size
         chunk_size = int(_CHUNK_SIZE / sze)
 
+        # Used by the forward pointers implementation
+        shas_to_ids, matched_segment_ids = set(), set()
+
         previous_shas = []
-        if previous_version:
-            previous_shas = set([Binary(x['sha']) for x in
-                                 collection.find({'symbol': symbol},
-                                                 projection={'sha': 1, '_id': 0},
-                                                 )
-                                 ])
+        if ARCTIC_FORWARD_POINTERS is FwPointersCfg.DISABLED:
```

Comment:
Again shouldn’t this be based on precious_version, not on the environment variable?

I’m worried that people / code writing to different libraries with different env variables set will leave the versions in a mess?

Reply:
I have already updated in my last commit this, be based whenever makes sense on the previous version configuration.